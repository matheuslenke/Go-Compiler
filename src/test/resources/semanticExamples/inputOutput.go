package sample

import "fmt"

func func1(){
	fmt.Println("testando print");

	var s string

	fmt.Scanln(s)

	fmt.Println(s)
}