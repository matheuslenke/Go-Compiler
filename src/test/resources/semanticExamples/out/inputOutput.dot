digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="import_spec"];
node2[label="Println@1"];
node3[label="parameters"];
node4[label="(string) arg@2"];
node3 -> node4;
node2 -> node3;
node5[label="result"];
node2 -> node5;
node1 -> node2;
node6[label="Scanln@3"];
node7[label="parameters"];
node8[label="(string) arg@4"];
node7 -> node8;
node6 -> node7;
node9[label="result"];
node6 -> node9;
node1 -> node6;
node0 -> node1;
node10[label="(func) func1@5"];
node11[label="parameters"];
node10 -> node11;
node12[label="block"];
node13[label="write"];
node14[label="Println@1"];
node13 -> node14;
node15[label="arguments"];
node16[label="(string) string@0"];
node15 -> node16;
node13 -> node15;
node12 -> node13;
node17[label="var_list"];
node18[label="(string) s@6"];
node17 -> node18;
node12 -> node17;
node19[label="read"];
node20[label="Scanln@3"];
node19 -> node20;
node21[label="arguments"];
node22[label="(string) s@6"];
node21 -> node22;
node19 -> node21;
node12 -> node19;
node23[label="write"];
node24[label="Println@1"];
node23 -> node24;
node25[label="arguments"];
node26[label="(string) s@6"];
node25 -> node26;
node23 -> node25;
node12 -> node23;
node10 -> node12;
node0 -> node10;
}
