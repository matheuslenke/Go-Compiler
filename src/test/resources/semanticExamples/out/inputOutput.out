[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:
Entry 0: testando print




Variables table:
Entry 0 -- name: fmt, line: 3, type: no_type, scopeName: global
Entry 1 -- name: Println, line: 3, type: no_type, scopeName: global
Entry 2 -- name: arg, line: 3, type: string, scopeName: Println
Entry 3 -- name: Scanln, line: 3, type: no_type, scopeName: global
Entry 4 -- name: arg, line: 3, type: string, scopeName: Scanln
Entry 5 -- name: func1, line: 5, type: func, scopeName: global
Entry 6 -- name: s, line: 8, type: string, scopeName: func1




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.836 s
[INFO] Finished at: 2022-02-24T20:53:55-03:00
[INFO] ------------------------------------------------------------------------
